packages/react/src/components/Switch/Switch.js

```diff
@@ -24,6 +25,14 @@ const Switch = props => {
     ...other
   } = props;
 
+  const tabRef = useRef(null);
+
+  useEffect(() => {
+    if (selected && tabRef.current) {
+      tabRef.current.focus();
```

Comment:
Don't want to change focus upon programmatic API call (though change in user-focusable state (at L63) makes sense). Change in focus needs to be associated with user gesture.

Reply:
According to #2488
>  The Right and Left Arrow keys should move focus forward and backward through the tabs.

Probably misunderstanding what you're saying here though.

Reply:
@dakahn You are right that left/arrow key should move focus. Such focus move shouldn't happen upon programmatic change in `selected` prop, though.

Reply:
@dakahn for example, `focus` is called in the `onClick` handler: https://reactjs.org/docs/hooks-reference.html#useref

Reply:
Sorry, circling back on this now -- so is the problem that I'm calling .focus in useEffect? @joshblack

Reply:
@dakahn actually, I think you have it right. `useEffect` is synchronizing the `selected` prop with the `tabRef` 👍

Maybe we could add another check to it? Like:

```js
  useEffect(() => {
    if (selected && document.activeElement !== tabRef.current) {
      tabRef.current.focus();
    }
  }, [selected]);
```

That way it's only focusing if it's not already the current active element

Reply:
With this approach; What happens if application changes selected `<Switch>` e.g. when new data arrives?

Reply:
@asudoh would we need to add a blur call in the callback to useEffect? e.g.

```js
useEffect(() => {
    if (selected && document.activeElement !== tabRef.current) {
      tabRef.current.focus();
    }
    () => tabRef.current.blur();
  }, [selected]);
```

Reply:
Given `selected` prop change happens upon programmatic change in addition to user gesture, my suggestion is distinguishing focus behavior from `select` prop. One way I can think of is wrapping `<Switch>` with `createRef()` so the keyboard action in `<ContentSwitcher>` can focus on selected `<Switch>`. Another way I can think of is adding a (function) prop to `<Switch>` that focuses on itself when it's called.

Reply:
What would that look like in code?

Reply:
For example, it can be:

```diff
diff --git a/packages/react/src/components/ContentSwitcher/ContentSwitcher.js b/packages/react/src/components/ContentSwitcher/ContentSwitcher.js
index c8ad8e2b1..2504aeb37 100644
--- a/packages/react/src/components/ContentSwitcher/ContentSwitcher.js
+++ b/packages/react/src/components/ContentSwitcher/ContentSwitcher.js
@@ -17,6 +17,13 @@ const { prefix } = settings;
 export default class ContentSwitcher extends React.Component {
   state = {};
 
+  /**
+   * The DOM references of child `<Switch>`.
+   * @type {Array<Element>}
+   * @private
+   */
+  _switchRefs = [];
+
   static propTypes = {
     /**
      * Pass in Switch components to be rendered in the ContentSwitcher
@@ -55,6 +62,10 @@ export default class ContentSwitcher extends React.Component {
   }
 
   getChildren(children) {
+    const switchRefs = React.Children.map(children, (child, index) => el => {
+      // Sets DOM ref of `<Switch>` to the corresponding index in `this._switchRefs`
+      this._switchRefs[index] = el;
+    }) || [];
     return React.Children.map(children, (child, index) =>
       React.cloneElement(child, {
         index,
@@ -64,6 +75,7 @@ export default class ContentSwitcher extends React.Component {
         ]),
         onKeyDown: this.handleChildChange,
         selected: index === this.state.selectedIndex,
+        ref: switchRefs[index],
       })
     );
   }
@@ -76,16 +88,23 @@ export default class ContentSwitcher extends React.Component {
     const { key } = data;
 
     if (key) {
+      const nextIndex = getNextIndex(
+        key,
+        selectedIndex,
+        this.props.children.length
+      );
       this.setState({
-        selectedIndex: getNextIndex(
-          key,
-          selectedIndex,
-          this.props.children.length
-        ),
+        selectedIndex: nextIndex,
+      }, () => {
+        const switchRef = this._switchRefs[nextIndex];
+        switchRef && switchRef.focus();
       });
     } else {
       if (selectedIndex !== index) {
-        this.setState({ selectedIndex: index });
+        this.setState({ selectedIndex: index }, () => {
+          const switchRef = this._switchRefs[index];
+          switchRef && switchRef.focus();
+        });
       }
     }
 
diff --git a/packages/react/src/components/Switch/Switch.js b/packages/react/src/components/Switch/Switch.js
index d44345720..98b9c68d6 100644
--- a/packages/react/src/components/Switch/Switch.js
+++ b/packages/react/src/components/Switch/Switch.js
@@ -6,14 +6,14 @@
  */
 
 import PropTypes from 'prop-types';
-import React, { useEffect, useRef } from 'react';
+import React from 'react';
 import classNames from 'classnames';
 import { settings } from 'carbon-components';
 import { matches, keyCodes } from '../../tools/key';
 
 const { prefix } = settings;
 
-const Switch = props => {
+const Switch = React.forwardRef(function Switch(props, tabRef) {
   const {
     className,
     index,
@@ -25,14 +25,6 @@ const Switch = props => {
     ...other
   } = props;
 
-  const tabRef = useRef(null);
-
-  useEffect(() => {
-    if (selected && document.activeElement !== tabRef.current) {
-      tabRef.current.focus();
-    }
-  }, [selected, tabRef.current]);
-
   const handleClick = e => {
     e.preventDefault();
     onClick({ index, name, text });
@@ -67,7 +59,7 @@ const Switch = props => {
       <span className={`${prefix}--content-switcher__label`}>{text}</span>
     </button>
   );
-};
+});
 
 Switch.propTypes = {
   /**
```